### PR TITLE
:bug: *: update listers to have correct NotFound messages

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Fixes https://github.com/kcp-dev/kcp/issues/2679

/assign @ncdc

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -32,21 +32,21 @@ require (
 	go.etcd.io/etcd/client/pkg/v3 v3.5.4
 	go.etcd.io/etcd/server/v3 v3.5.0
 	go.uber.org/multierr v1.7.0
-	golang.org/x/net v0.0.0-20220722155237-a158d28d115b
+	golang.org/x/net v0.5.0
 	google.golang.org/genproto v0.0.0-20220519153652-3a47de7e79bd
 	gopkg.in/square/go-jose.v2 v2.2.2
-	k8s.io/api v0.24.3
+	k8s.io/api v0.26.1
```
Comment (ncdc) [inline go.mod:38]: Wouldn't have expected all these bumps?

Comment (stevekuznetsov) [reply go.mod:38]: huh. I just attempted to bump the client-go bits. sec.

Comment (stevekuznetsov) [reply go.mod]: Oops, did a `go get -u ...` ... fixed!

**Review by ncdc — APPROVED** [review]

**Comment by stevekuznetsov** [comment]
https://github.com/kcp-dev/kcp/issues/2685

**Comment by stevekuznetsov** [comment]
@ncdc needed to `go mod tidy`, would appreciate a re-tag

**Review by ncdc — APPROVED** [review]

**Pull request merged**